flow/hooks/log_operations.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2018 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Builtin execution hook for basic logging."""
+import logging
+
+
+class LogOperations:
+    """:class:`~.LogOperations` logs information about the execution of operations to a log file.
+
+    This hooks provides information, optionally, on the start, successful completion, and/or
+    erroring of one or more operations in a `flow.FlowProject` instance. The logs are stored in a
+    file given by the parameter ``fn_logfile``. This file will be appended to if it already exists.
+
+    The default formating for the log provides the time, job id, log level, and log message.
+
+    Examples
+    --------
+    The code block below provides an example of how install  :class:`~.LogOperations` to a
+    instance of :class:`~.FlowProject`
+
+    .. code-block:: python
+        from flow import FLowProject
+        from flow.hooks import LogOperations  # Import build
+
+
+    Parameters
+    ----------
+    fn_logfile: log filename
+        The name of the log file in the job workspace. Default is "execution-record.log".
+    """
+
+    def __init__(self, fn_logfile="execution-record.log"):
+        self._fn_logfile = fn_logfile
+        # getLogger keep its own cache. This just serves to reduce the time spent setting up loggers
+        # by only doing it once.
+        self._loggers = dict()
+
+    def on_start(self, operation, job):
+        """Log the start of execution of a given job(s) operation pair."""
+        self._get_logger(job).info(f"Starting execution of operation '{operation}'.")
+
+    def on_success(self, operation, job):
+        """Log the successful completion of a given job(s) operation pair."""
+        self._get_logger(job).info(
+            f"Successfully finished execution of operation '{operation}'."
+        )
+
+    def on_exception(self, operation, error, job):
+        """Log the raising of an error in the execution of a given job(s) operation pair."""
+        self._get_logger(job).info(
+            f"Execution of operation '{operation}' failed with error '{error}'."
+        )
+
+    def _get_logger(self, job):
+        if job not in self._loggers:
+            self._loggers[job] = self._setup_logger(job)
+        return self._loggers[job]
```

Comment:
I am not too sure about how hooks work, is the `job`  here the id of the job or `signac.Job` itself?

Reply:
This is the `signac.Job` itself